server/app/services/agent/message_handler.rb

```diff
@@ -188,13 +194,17 @@ def on_node_stat(grid, data)
       node = grid.host_nodes.find_by(node_id: data['id'])
       return unless node
       data = fixnums_to_float(data)
+      time = data['time'] ? Time.parse(data['time']) : Time.now.utc
```

Comment:
Agreed, it looks like both the agent host and container stats are missing any `time` field:

* https://github.com/kontena/kontena/blob/fix/hostnode_statistics_missing_timestamp/agent/lib/kontena/workers/node_info_worker.rb#L126
* https://github.com/kontena/kontena/blob/fix/hostnode_statistics_missing_timestamp/agent/lib/kontena/workers/stats_worker.rb#L109

The agent events may be queued up for long periods of time if there's e.g. websocket connection issues.

IMO either also fix the agent side as part of this PR, or file a separate issue for that.